src/cmd/logic-function.js

```diff
@@ -46,9 +52,129 @@ module.exports = class LogicFunctionsCommand extends CLICommandBase {
 		console.log(org, name, id);
 	}
 
-	async create({ org, params: { filepath } }) {
-		// TODO
-		console.log(org, filepath);
+	async create({ org, name, params : { filepath } } = { params: { } }) {
+		const orgName = getOrgName(org);
+		const api = createAPI();
+		// get name from filepath
+		if (!filepath) {
+			// use default directory
+			filepath = '.';
+		}
+		if (!name) {
+			const question = {
+				type: 'input',
+				name: 'name',
+				message: 'What would you like to call your Function?'
+			};
+			const result =  await this.ui.prompt([question]);
+			name = result.name;
+		}
+		// trim name
+		name = name.trim();
+		// ask for description
+		const question = {
+			type: 'input',
+			name: 'description',
+			message: 'Add a description for your Function (optional)'
+		};
+		const result =  await this.ui.prompt([question]);
+		const description = result.description;
+		const slugName = name.toLowerCase().replace(/\s/g, '-');
```

Comment:
This is the slugify function from api-service. Can you use that? Copy it to a helper
```
	/**
	 * Converts a string to the slug version by replacing
	 * spaces and underscores with dashes, changing
	 * to lowercase, and removing anything other than
	 * numbers, letters, and dashes
	 * @param  {String} str string to slugify
	 * @return {String}     slugified version of str
	 */
	slugify(str) {
		const slug = str.trim().toLowerCase()
			// replace every group of spaces and underscores with a single hyphen
			.replace(/[ _]+/g, '-')
			// delete everything other than letters, numbers and hyphens
			.replace(/[^a-z0-9-]/g, '');
		return slug;
	},
```

Reply:
Done here! 👉 https://github.com/particle-iot/particle-cli/pull/691/commits/348a5832b49b97b5bd6d5a9c33610b6d1d9b23b8